### PR TITLE
configuration changes

### PR DESCRIPTION
**Pull request by bheni**

### CODE REVIEW

File: go/cmd/dolt/commands/sqlserver/serverconfig.go [hunk 1]
```diff
@@ -25,140 +27,221 @@ type LogLevel string
...
 	LogLevel_Info    LogLevel = "info"
 	LogLevel_Warning LogLevel = "warning"
 	LogLevel_Error   LogLevel = "error"
 	LogLevel_Fatal   LogLevel = "fatal"
 )
 
+// String returns the string representation of the log level.
+func (level LogLevel) String() string {
+	switch level {
+	case LogLevel_Trace:
```
Comment (zachmu) [inline go/cmd/dolt/commands/sqlserver/serverconfig.go:41]: Can't you combine all these into one case statement?

File: go/cmd/dolt/commands/sqlserver/servercontroller.go [hunk 2]
```diff
@@ -19,8 +19,10 @@ import (
 )
 
 type ServerController struct {
```
Comment (bheni) [inline go/cmd/dolt/commands/sqlserver/servercontroller.go:21]: The way this was implemented was deadlocking some of the tests.  Specifically TestServerBadArgs would hang on the call to WaitForStart as the server never started and exited early.  Now if the server is stopped without ever being started WaitForStart will return.

File: go/cmd/dolt/commands/sqlserver/serverconfig.go [hunk 3]
```diff
@@ -25,140 +27,221 @@ type LogLevel string
...
+	}
+	if config.LogLevel().String() == "unknown" {
+		return fmt.Errorf("loglevel is invalid: %v\n", string(config.LogLevel()))
+	}
+	return nil
+}
+
+// ConnectionString returns a Data Source Name (DSN) to be used by go clients for connecting to a running server.
+func ConnectionString(config ServerConfig) string {
+	return fmt.Sprintf("%v:%v@tcp(%v:%v)/dolt", config.User(), config.Password(), config.Host(), config.Port())
```
Comment (zachmu) [inline go/cmd/dolt/commands/sqlserver/serverconfig.go:240]: /dolt is wrong for most configs Either pick a DB that exists, or don't name one.

File: go/cmd/dolt/commands/sqlserver/server_test.go [hunk 4]
```diff
@@ -76,7 +76,6 @@ func TestServerBadArgs(t *testing.T) {
 		{"-P", "300"},
 		{"-P", "90000"},
 		{"-u", ""},
-		{"-t", "-1"},
```
Comment (bheni) [inline go/cmd/dolt/commands/sqlserver/server_test.go:79]: Had to remove this test case as it now fails earlier. The call to Serve is never made inside of startServer so serverController.WaitForServe will never return a start error.

Comment (bheni) [reply go/cmd/dolt/commands/sqlserver/server_test.go:79]: ... it still fails.  just not in this way.

File: go/cmd/dolt/commands/sqlserver/yaml_config.go [hunk 5]
```diff
@@ -0,0 +1,124 @@
...
+
+type DatabaseYAMLConfig struct {
+	Name string
+	Path string
+}
+
+type ListenerYAMLConfig struct {
+	HostStr            string `yaml:"host"`
+	PortNumber         int    `yaml:"port"`
+	MaxConnectoins     uint64 `yaml:"max_connections"`
```
Comment (zachmu) [inline go/cmd/dolt/commands/sqlserver/yaml_config.go:41]: Typo

File: go/cmd/dolt/commands/sqlserver/serverconfig.go (hunk 1 above)
Comment (bheni) [reply go/cmd/dolt/commands/sqlserver/serverconfig.go:41]: Yeah, I was just matching style and going for a minimal change... but I do hate the way this is written.

File: go/cmd/dolt/commands/sqlserver/yaml_config.go (hunk 5 above)
Comment (bheni) [reply go/cmd/dolt/commands/sqlserver/yaml_config.go:41]: fixed

File: go/cmd/dolt/commands/sqlserver/serverconfig.go (hunk 3 above)
Comment (bheni) [reply go/cmd/dolt/commands/sqlserver/serverconfig.go]: fixed

**Pull request merged**